plugins/web/opentelemetry-instrumentation-document-load/src/utils.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span } from '@opentelemetry/api';
+import { otperformance } from '@opentelemetry/core';
+import {
+  hasKey,
+  PerformanceEntries,
+  PerformanceLegacy,
+  PerformanceTimingNames as PTN,
+} from '@opentelemetry/web';
+import { EventNames } from './enums/EventNames';
+
+export const getPerformanceNavigationEntries = (): PerformanceEntries => {
+  const entries: PerformanceEntries = {};
+  const performanceNavigationTiming = ((otperformance as unknown) as Performance).getEntriesByType?.(
+    'navigation'
+  )[0] as PerformanceEntries;
+
+  if (performanceNavigationTiming) {
+    const keys = Object.values(PTN);
+    keys.forEach((key: string) => {
+      if (hasKey(performanceNavigationTiming, key)) {
+        const value = performanceNavigationTiming[key];
+        if (typeof value === 'number') {
+          entries[key] = value;
+        }
+      }
+    });
+  } else {
+    // // fallback to previous version
+    const perf: typeof otperformance & PerformanceLegacy = otperformance;
+    const performanceTiming = perf.timing;
+    if (performanceTiming) {
+      const keys = Object.values(PTN);
+      keys.forEach((key: string) => {
+        if (hasKey(performanceTiming, key)) {
+          const value = performanceTiming[key];
+          if (typeof value === 'number') {
+            entries[key] = value;
+          }
+        }
+      });
+    }
+  }
+
+  return entries;
+};
+
+const performancePaintNames = {
+  'first-paint': EventNames.FIRST_PAINT,
```

Comment:
I don't see a lot of dashes in names/semantic conventions in otel... `.` and `_` yes, but not `-`.  Perhaps the camel case names are fine as they are?  Most other browser `performance` event names are passed through unchanged.

Reply:
So API defines them as `first-paint` and I proposed to use `firstPaint`.
The `EventNames` enum is for our OT enum names and here I define how to map them.
Other events are camel cased and I wanted to follow this convention.

Reply:
Oh, I was reading it backwards!  Apologies!

Reply:
@johnbley does it mean you are ok with this changes then ?

Reply:
Yes, it looks good.  Sorry I didn't officially, approve, will do so.